src/parsimmon.js

```diff
@@ -494,6 +497,30 @@
     });
   }
 
+  function lookahead(x) {
+    if (typeof x === 'string') {
+      // similar to string, but does not consume
+      return Parsimmon(function(input, i) {
+        var j = i + x.length;
+        var head = input.slice(i, j);
+        if (head === x) {
+          return makeSuccess(i, '');
+        } else {
+          return makeFailure(i, x);
+        }
+      });
+    } else if (x instanceof RegExp) {
+      var regexp = RegExp('^(?:' + x.source + ')', flags(x));
```

Comment:
See `assertRegexp`, certain regexp flags are not allowed in Parsimmon, so we should use that function here.